homeassistant/components/cover/mqtt.py

```diff
@@ -67,30 +92,48 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         config.get(CONF_PAYLOAD_STOP),
         config.get(CONF_OPTIMISTIC),
         value_template,
+        config.get(CONF_TILT_OPEN_POSITION),
+        config.get(CONF_TILT_CLOSED_POSITION),
+        config.get(CONF_DEVICE_CLASS),
+        config.get(CONF_TILT_MIN),
+        config.get(CONF_TILT_MAX),
     )])
 
 
 class MqttCover(CoverDevice):
     """Representation of a cover that can be controlled using MQTT."""
 
-    def __init__(self, name, state_topic, command_topic, qos, retain,
-                 state_open, state_closed, payload_open, payload_close,
-                 payload_stop, optimistic, value_template):
+    def __init__(self, name, state_topic, command_topic, tilt_command_topic,
+                 tilt_status_topic, qos, retain, state_open, state_closed,
+                 payload_open, payload_close, payload_stop,
+                 optimistic, value_template, tilt_open_position,
+                 tilt_closed_position, device_class, tilt_min, tilt_max):
         """Initialize the cover."""
+        if tilt_command_topic is not None:
```

Comment:
The method in the cover looks at the current_cover_tilt_value to return the supported features. If the tilt support isn't there then there's no way to set the tilt_status if it's not persisted on the queue. Chicken and egg situation